lib/common.go

```diff
@@ -26,18 +26,44 @@ const (
 // the slave knows from which string permutation to start and at which string
 // permutation to stop.
 type Task struct {
-	ID        int
-	JobID     int
-	Algorithm string
-	Salt      string
-	Digest    string
-	CharSet   string
-	Length    int
-	Start     string
-	End       string
+	Job
+	ID      int
+	Start   []byte
+	TaskLen int
 }
 
 // ToJSON serializes a Task to JSON.
 func (t *Task) ToJSON() ([]byte, error) {
 	return json.Marshal(t)
 }
+
+type Algorithm int
+
+const (
+	PBKDF2 Algorithm = iota
+	BCRYPT
+	SCRYPT
+	ARGON2
+)
+
+type CharSet int
```

Comment:
I just thought that `CharSet` might not be the right name. In fact, charset normally indicates stuff like UTF-8 and such. Maybe just `Chars`?